moveit_planners/ompl/ompl_interface/include/moveit/ompl_interface/planning_context_manager.h

```diff
@@ -209,9 +218,9 @@ class PlanningContextManager
   void registerPlannerAllocatorHelper(const std::string& planner_id);
 
   /** \brief This is the function that constructs new planning contexts if no previous ones exist that are suitable */
-  ModelBasedPlanningContextPtr getPlanningContext(const planning_interface::PlannerConfigurationSettings& config,
-                                                  const StateSpaceFactoryTypeSelector& factory_selector,
-                                                  const moveit_msgs::MotionPlanRequest& req) const;
+  ModelBasedPlanningContextPtr createPlanningContext(const planning_interface::PlannerConfigurationSettings& config,
```

Comment:
Why the name change? This method will actually return a planning context from a cache if one is available. In that case no new planning context is created.

Reply:
The name change probably only made sense to me personally, I'll revert it.

I think it can be confusing that both methods do part of the construction of the planning context. For examples, after the context is created, or a cached context is retrieved, `context.clear()` is called. And both methods call a bunch of `context->setXXX(..)` methods.

And then as a cherry on top of the cake, `context->configure()` is called to do even more configuration :)

I think it can be confusing to have to the configuration of the context so spread out in the code base, without clear names indications what is done where. But I don't have a solution...